### PR TITLE
[Jira] Map custom field ids with their names

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Part of https://github.com/elastic/connectors/issues/2415

Previously we're getting custom field ids in the API response, now we fetch all custom fields and mapping their names with the ids present in the response.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Release Notes

Maps custom fields' IDs of `issue` documents to the custom field names.

### CODE REVIEW

File: connectors/sources/jira.py [hunk 1]
```diff
@@ -934,6 +949,9 @@ async def _get_issues(self, custom_query=""):
             if self.jira_client.projects == [WILDCARD]
             else projects_query
         )
+
+        self.custom_fields = await self.jira_client.get_jira_fields()
```
Comment (wangch079) [inline connectors/sources/jira.py:953]: Can we move this to `get_docs`? `_get_issues` can be called multiple times

**Review by wangch079 — APPROVED** [review]
LGTM, just one nit

File: connectors/sources/jira.py (hunk 1 above)
Comment (moxarth-elastic) [reply connectors/sources/jira.py]: Yes, that's right! Moved to `get_docs`.

**Review by wangch079 — APPROVED** [review]

**Comment by Gleb-Gadyatskiy** [comment]
Got this issue after this PR changes https://github.com/elastic/connectors/issues/2415#issuecomment-2089298705

**Pull request merged**